### PR TITLE
Consolidate tox.ini and pyproject.toml

### PR DESCRIPTION
**Pull request by bittner**
Just a little cleanup and modernization of `tox.ini` with moving some configuration bits to `pyproject.toml`.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -42,41 +55,25 @@ commands = flake8 {posargs}
...
 
 [testenv:readme]
 description = Ensure README renders on PyPI
-deps = twine
+deps =
+    build
+    twine
 commands =
-    {envpython} setup.py -q sdist bdist_wheel
+    python -m build
```
Comment (smithdc1) [inline tox.ini:65]: Nice 👍

**Review by smithdc1 — APPROVED** [review]

**Comment by smithdc1** [comment]
Looks good to me. Looking at this I realise we should likely drop Py3.6 and Django2.2 (it's eol in less than a month).

**Comment by bittner** [comment]
> Looking at this I realise we should likely drop Py3.6 and Django2.2 (it's eol in less than a month).

Good point! Let's be good though to people that love sticking to legacy software (probably 95% of commercial users) and give them time to migrate. Haha. :smile:

**Comment by bittner** [comment]
> Could do with getting a release out so there's a release with all the features of the past couple of years before dropping it?

Excellent point! We only need to update the package metadata (probably on the [version number](https://github.com/jazzband/django-analytical/blob/main/analytical/__init__.py#L7)) and the [change log](https://github.com/jazzband/django-analytical/blob/main/CHANGELOG.rst), and push a version tag. That should do it. – Do you want to go about it?

Not sure about the new version number. Did we have breaking changes, or was it mainly reformatting (Black) and refactoring (UnitTest ➜ pytest)?  ([changes since v3.0.0](https://github.com/jazzband/django-analytical/compare/v3.0.0..main))

**Pull request merged**